demo-app/src/lib/mysql-uqi-client.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  Iterator,
+  uqi,
+  type UqiClient,
+  type UqiColumn,
+  type UqiContext,
+  type UqiMappedType,
+  type UqiResult,
+} from '@open-truss/open-truss'
+import mysql, { type Connection, type FieldPacket } from 'mysql2/promise'
+import { URL } from 'url'
+
+export interface MysqlConfig {
+  uri: string
+  socketPath?: string
+}
+
+function makeUqiColumnCompatible(fields: FieldPacket[]): UqiColumn[] {
+  return fields.map((field) => {
+    return {
+      name: String(field.name),
+      type: String(field.type),
+    }
+  })
+}
+
+export default async function (config: MysqlConfig): Promise<UqiClient> {
+  const parsedUrl = new URL(config.uri)
+
+  const typeMappings: Record<string, UqiMappedType> = {
```

Comment:
Would it work to have this a constant? might make this function a bit easier to read